pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -157,6 +165,13 @@ type PriorityQueue struct {
 
 	cond sync.Cond
 
+	// inFlightPods holds the UID of all pods which have been popped out for which Done
+	// hasn't been called yet - in other words, all pods that are currently being
+	// processed. For each of those, it stores PodInfo.MoveRequestCycle (if one was observed)
+	// and the pod object. This is intentionally not a PodInfo pointer because accessing
```

Comment:
This comment comes from https://github.com/kubernetes/kubernetes/pull/115796/files#diff-719907c9be2fdbb70be08473fa5f86fd3c42ebc3c0fd2f347065390bec6b26c5 but it's not true anymore because you changed `inFlightPod.pod` from `*v1.Pod` to `*framework.QueuedPodInfo`.

This is problematic, see my other comment about locking in `runRequeuePlugins`.

Reply:
Asking about the validity of that particular usage of `unschedulablePlugins` was next on my list - now I don't need to bring it up :grin:

But the new logic also needs better explanations.